flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -155,55 +199,79 @@ public GetRecordsResult getRecords(String shardIterator, int maxRecordsToGet) {
 	 * @param startingSeqNum the sequence number that the iterator will start from
 	 * @return the shard iterator
 	 */
+	@Override
 	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, String startingSeqNum) {
-		return kinesisClient.getShardIterator(shard.getStreamName(), shard.getShardId(), shardIteratorType, startingSeqNum).getShardIterator();
+		return kinesisClient.getShardIterator(shard.getStreamName(), shard.getShard().getShardId(), shardIteratorType, startingSeqNum).getShardIterator();
```

Comment:
Okay. Actually, the 5 rps limit for `getShardIterator()` is per shard. We'll only have one thread making this call for each shard, so the rate limit won't be a problem for us. But as explained above, we should still retry + fail hard to let the user know if they need to adjust other non-Flink consumers.